UniversalDashboard.Forge.psm1

```diff
@@ -117,36 +117,44 @@ function New-UDDesktopApp {
         if ($PathInfo.PSIsContainer)
         {
             Write-Verbose "Copying contents of $Path to $src"
-            Copy-Item -Path "$($PathInfo.FullName)/*" -Destination $src -Container -Recurse
+            Copy-Item -Path "$($PathInfo.FullName)/*" -Destination $src -Exclude (Split-Path $OutputPath -Leaf) -Recurse
         }
 
         Write-Verbose "Copying dashboard and index.js to electron src folder: $src"
 
-        $Content = Get-Content $Dashboard -Raw
-        $Content = "
-        `$Env:PSModulePath = `$Env:PSModulePath + `";`$PSScriptRoot`"
-        Import-Module UniversalDashboard" + [System.Environment]::NewLine + $Content
-        $Content | Out-File (Join-Path $Src "dashboard.ps1") -Force -Encoding utf8
+        $content = Get-Content $dashboard -Raw
+        $content = @"
+{0} = {0} + "; {1}"
+Import-Module UniversalDashboard
+
+{2}
+"@ -f '$Env:PSModulePath', '$PSScriptRoot', $content
+
+        $content | Out-File (Join-Path $src "dashboard.ps1") -Force -Encoding utf8
 
         Copy-Item -Path (Join-Path $PSScriptRoot "index.js" ) -Destination $src -Force
-        $IndexJs = Join-Path $src "index.js"
+        $indexJs = Join-Path $src "index.js"
 
-        $port = Get-PortNumber -Path $Dashboard
-        Set-ForgeVariable -IndexPath $IndexJs -PowerShellHost $PowerShellHost -Port $port
+        $port = Get-PortNumber -Path $dashboard
+        Set-ForgeVariable -IndexPath $indexJs -PowerShellHost $PowerShellHost -Port $port
 
-        $PackageConfig = [IO.Path]::Combine($OutputPath, $Name, 'package.json')
-        $SquirrelSplat = @{'ConfigPath' = $PackageConfig}
-        if ($IconUrl) {$SquirrelSplat['IconUrl'] = $IconUrl}
-        if ($SetupIcon) {$SquirrelSplat['SetupIcon'] = $SetupIcon}
-        if ($LoadingGif) {$SquirrelSplat['LoadingGif'] = $LoadingGif}
-        Set-SquirrelConfig @SquirrelSplat
+        $packageConfig = [IO.Path]::Combine($OutputPath, $Name, 'package.json')
+        $squirrelSplat = @{'ConfigPath' = $packageConfig}
+        if ($IconUrl) {$squirrelSplat['IconUrl'] = $IconUrl}
+        if ($SetupIcon) {
+            $iconPath = (Get-ChildItem -Path $src -Filter (Split-Path $SetupIcon -Leaf) -Recurse).FullName
+            $squirrelSplat['SetupIcon'] = $iconPath
+        }
+        if ($LoadingGif) {
+            $gifPath = (Get-ChildItem -Path $src -Filter (Split-Path $LoadingGif -Leaf) -Recurse).FullName
+            $squirrelSplat['LoadingGif'] = $gifPath
+        }
```

Comment:
When building the package.json file I kept on getting errors when the application was getting built. The ico or gif file could never be located. I found they had to be in reference to the $src directory. I added the lines to find the path name of where those files exist in the source directory.

Reply:
If anyone has a better, albeit cleaner, way of doing this please let me know.